RoslynPluginGenerator/AnalyzerPluginGenerator.cs

```diff
@@ -61,27 +61,97 @@ public bool Generate(ProcessedArgs args)
                 throw new ArgumentNullException("args");
             }
 
-            IPackage package = this.packageHandler.FetchPackage(args.PackageId, args.PackageVersion);
-            if (package == null)
+            IPackage targetPackage = this.packageHandler.FetchPackage(args.PackageId, args.PackageVersion);
+            IEnumerable<IPackage> dependencyPackages = this.packageHandler.GetInstalledDependencies(targetPackage);
+
+            if (targetPackage == null)
             {
                 return false;
             }
 
-            string packageDir = this.packageHandler.GetLocalPackageRootDirectory(package);
-            IEnumerable<DiagnosticAnalyzer> analyzers = GetAnalyzers(packageDir, this.packageHandler.LocalCacheRoot, args.Language);
-            if (!analyzers.Any())
+            // Check that there are analyzers in the target package from which information can be extracted
+            if (!HasAnalyzers(targetPackage, args.Language))
             {
-                return false;
+                this.logger.LogWarning(UIResources.APG_NoAnalyzersFound, targetPackage.Id);
+
+                if (!args.RecurseDependencies || !dependencyPackages.Any())
+                {
+                    return false;
+                }
+                else
+                {
+                    // Possible sub-case - target package has dependencies that contain analyzers
+                    bool dependencyPackagesHaveAnalyzers = false;
+                    foreach (IPackage dependencyPackage in dependencyPackages)
+                    {
+                        bool dependencyPackageHasAnalyzers = HasAnalyzers(dependencyPackage, args.Language);
+
+                        if (!dependencyPackageHasAnalyzers)
+                        {
+                            this.logger.LogWarning(UIResources.APG_NoAnalyzersFound, dependencyPackage.Id);
+                        }
+
+                        dependencyPackagesHaveAnalyzers = dependencyPackagesHaveAnalyzers || dependencyPackageHasAnalyzers;
+                    }
+
+                    if (!dependencyPackagesHaveAnalyzers)
+                    {
+                        return false;
+                    }
+                }
             }
 
-            IEnumerable<IPackage> licenseAcceptancePackages = this.GetPackagesRequiringLicenseAcceptance(package);
-            if (licenseAcceptancePackages.Any() && ! args.AcceptLicenses)
+            // Check for packages that require the user to accept a license
+            IEnumerable<IPackage> licenseAcceptancePackages = this.GetPackagesRequiringLicenseAcceptance(targetPackage);
+            if (licenseAcceptancePackages.Any() && !args.AcceptLicenses)
             {
-                this.logger.LogError(UIResources.APG_NGPackageRequiresLicenseAcceptance, package.Id, package.Version.ToString());
+                this.logger.LogError(UIResources.APG_NGPackageRequiresLicenseAcceptance, targetPackage.Id, targetPackage.Version.ToString());
```

Comment:
The list of packages requiring acceptance is not longer strictly accurate e.g. in the case of _CodeCracker_ user will be told they have accepted the licenses for _CodeCracker_, _CodeCracker.VisualBasic_, and _CodeCracker.CSharp_, when in fact the only plugin that will be generated is _CodeCracker.CSharp_.
I don't think this is worth fixing now but it should at least be documented in the code.